examples/server/server.cpp

```diff
@@ -2790,15 +2799,118 @@ int main(int argc, char **argv)
                                 {"system_info", llama_print_system_info()},
                             });
 
-    // load the model
-    if (!llama.load_model(params))
+    httplib::Server svr;
+
+    ServerState server_state = LOADING_MODEL;
+
+    svr.set_default_headers({{"Server", "llama.cpp"},
+                             {"Access-Control-Allow-Origin", "*"},
+                             {"Access-Control-Allow-Headers", "content-type"}});
+
+    svr.Get("/health", [&](const httplib::Request&, httplib::Response& res) {
+        switch(server_state) {
```

Comment:
There is a data race on `server_state`. Although not an issue in reality, the thread sanitizer is not going to like this, so lets make it atomic